src/main/scala/com/evolutiongaming/prometheus/ObserveDuration.scala

```diff
@@ -1,65 +1,160 @@
 package com.evolutiongaming.prometheus
 
-import io.prometheus.client.Collector
+import io.prometheus.client.{Collector, SimpleTimer}
 
+import scala.annotation.{nowarn, unused}
 import scala.concurrent.{ExecutionContext, Future}
-
-trait ObserveDuration[F] {
-
+import scala.math.Numeric.Implicits.*
+import scala.util.Try
+
+/** Time duration measurement syntax for [[HasObserve]]-kind metrics from the prometheus client, i.e., Summary, Histogram.
+  *
+  * Here time is always reported in seconds, which means your prometheus metric name should end in `_seconds`.
+  *
+  * The class is not supposed to be used directly but as an implicit syntax provided by [[PrometheusHelper]].
+  */
+sealed trait ObserveDuration[F] {
+
+  /** Measures evaluation time of a block in seconds with nano-time precision
+    */
   def timeFunc[T](f: => T): T
-  def timeFuncNanos[T](f: => T): T
 
+  @deprecated(
+    message = "use timeFunc instead - it has nano-time precision now",
+    since = "1.1.0"
+  )
+  def timeFuncNanos[T](f: => T): T = timeFunc(f)
+
+  /** Measures evaluation time of an asynchronous block in seconds with nano-time precision
+    */
   def timeFuture[T](f: => Future[T]): Future[T]
-  def timeFutureNanos[T](f: => Future[T]): Future[T]
 
+  @deprecated(
+    message = "use timeFuture instead - it has nano-time precision now",
+    since = "1.1.0"
+  )
+  def timeFutureNanos[T](f: => Future[T]): Future[T] = timeFuture(f)
+
+  @deprecated(
+    message = "use timeTillNowMillis(: Long) with a primitive arg type and explicit precision name suffix",
+    since = "1.1.0"
+  )
   def timeTillNow[T](start: T)(implicit numeric: Numeric[T]): Unit
```

Comment:
Could you pls add doc that explicitly says that the `start` expected to be in milliseconds (if i got the implementation right)

Reply:
Yes, np. I was on the fence on whether to add scaladoc's to deprecated methods here - users will get the deprecation warning with a clue to the new documented one. I will add the scaladocs to deprecated methods in the class.